sw/device/lib/dif/README.md

```diff
@@ -50,7 +50,144 @@ memory-mapped hardware, and modifying processor CSRs.
   DIFs must not call DIFs in other DIF libraries.
 * DIF library headers must be polyglot headers for C and C++.
 
-### DIF Guidance
+### DIF API Guidance
+
+The following rules specify the basic API that each DIF must conform to. These
+rules specify the names of types, constants, and functions that each DIF must
+define for providing certain kinds of non-device-specific functionality (such as
+initializing handles or managing interrupts).
+
+Notational caveats:
+* The token `<ip>` is the name of a peripheral as given in its comportable
+specification, in `snake_case` or `PascalCase` as is appropriate.
```

Comment:
Comment: We haven't yet mandated that the DIF name exactly matches the comportable IP name, though this is what we want going forwards. @gkelly thoughts?

Reply:
Personally, I'm not the biggest fan, simply because names like `pwrmgr` are... not readable. I still think this is the direction we're moving towards and we should just keep going instead of having another place where we need to make decisions.

Reply:
Let's call it "the short IP name", and come back to this later